penman/__main__.py

```diff
@@ -222,12 +222,16 @@ def main():
         '--indicate-branches', action='store_true',
         help='insert triples to indicate tree structure')
 
+    norm.add_argument(
+        '--encoding', action='store_true', default=locale.getpreferredencoding(),
+        help='encoding to use for input/output. Defaults to system default')
+
```

Comment:
Oh, I am sorry. Did this a bit too quickly I am afraid. I've now added it to the parser and removed store_true. Type checking is not necessary I think: Python will throw an error when opening a file anyway if it is the unexpected type, so we should not take on that responsible of type checking _again_.

I tend to use getpreferredencoding because it has the benefit, in my opinion, that when users run `penman -h` and when a [default argument helper](https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter) is used, they will see what the current default encoding is. It gives users a bit more information than just None. But considering that you are currently not using a default args formatter, that is a non-argument. So I changed all default values to None as requested.

Reply:
I see. The point of the type checking is that `ArgumentParser` checks for `ValueError`, `TypeError`, and `ArgumentError` and shows a more helpful error message, e.g., the command usage.

Your arguments for using the `getpreferredencoding()` are good, but as you say I'm not using the default argument helper. Another concern is that the function is not thread safe with the default value of `True` for the `do_setlocale` parameter. Not sure this would make a difference for anyone running Penman in a multithreaded setting, though.

Reply:
Got it! Removed, all references to locale.